### PR TITLE
Editor Store: Make refreshPost() effect bypass cache by using a new apiFetch option

### LINKED ISSUE #12055: Editor Store: refreshPost() returns cached post
**Describe the bug**
Contrarily to what its name suggests, `refreshPost` doesn't actually refetch post data from the network. Found here: https://github.com/Automattic/wp-calypso/issues/28664

**To Reproduce**
1. Edit a post
2. Open your browser console's 'Network' tab, and clear it
3. In the browser console, enter `wp.data.dispatch( 'core/editor' ).refreshPost()`
4. Go back to the 'Network' tab. Note that no network request has been fired.

**Reason**
This is happening because of `apiFetch`'s [`preloading` middleware](https://github.com/WordPress/gutenberg/blob/b198e5c4791b2282332def0de89841cd64d75f8e/packages/api-fetch/src/middlewares/preloading.js), which is used to preload and cache data for some REST API routes -- among them, posts. Specifically, try commenting out https://github.com/WordPress/gutenberg/blob/b198e5c4791b2282332def0de89841cd64d75f8e/lib/client-assets.php#L1075-L1079 and running through the above steps again -- verify that `refreshPost` then actually fires a network request.

**Additional context**
- Using Gutenberg 4.4.0.
- This is sort of a follow-up to #11879, related to work on Jetpack's Publicize Gutenberg extension (which is mostly happening in the Automattic/wp-calypso repo)

/cc @tyxla

### PR DESCRIPTION
**Pull request by ockham**
## Description
- apiFetch: Add `usePreloaded` option to allow bypassing the cache
- Editor Store: Make `refreshPost()` effect bypass cache (by setting `usePreloaded` to `true`)

## How has this been tested?
See  #12055 for instructions to reproduce the issue (on `master`).
Verify that on this branch, entering `wp.data.dispatch( 'core/editor' ).refreshPost()` into your browser console _does_ fire a network request.
Verify that any other network requests for routes that are preloaded still use cached data, i.e. don't fire network requests -- for a list, see https://github.com/WordPress/gutenberg/blob/b198e5c4791b2282332def0de89841cd64d75f8e/lib/client-assets.php#L1038-L1047 

## Types of changes
Bug fix -- fixes #12055

## Checklist
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 

/cc @tyxla @youknowriad

### CODE REVIEW

File: packages/editor/src/store/effects/posts.js [hunk 1]
```diff
@@ -313,6 +313,7 @@ export const refreshPost = async ( action, store ) => {
 	const postType = await resolveSelector( 'core', 'getPostType', postTypeSlug );
 	const newPost = await apiFetch( {
 		path: `/wp/v2/${ postType.rest_base }/${ post.id }?context=edit`,
+		usePreloaded: false,
```
Comment (youknowriad) [inline packages/editor/src/store/effects/posts.js]: The only thing that's bothering me is that the `editor` package is now aware of the existence of a potential preloading middleware.

No strong opinions but an alternative could be to add a timestamp to the query string.

Comment (ockham) [reply packages/editor/src/store/effects/posts.js]: I was considering using some more or less random qarg. I like the idea of a timestamp.

Comment (ockham) [reply packages/editor/src/store/effects/posts.js]: Is there precedent for the timestamp qarg? Just looking for possibly established nomenclature...

Comment (youknowriad) [reply packages/editor/src/store/effects/posts.js]: I'm not aware of any personally

Comment (ockham) [reply packages/editor/src/store/effects/posts.js]: Timestamp qarg PR: #12060

Comment (tyxla) [reply packages/editor/src/store/effects/posts.js]: This does the job, but probably the other PR looks a little better - I like the simplicity.

No matter which one we choose - let's make sure to document this caching feature thoroughly.

**Comment by youknowriad** [comment]
closing as superseded by #12060

**Pull request closed without merging**